contracts/GroupHealth.sol

```diff
@@ -0,0 +1,297 @@
+        manager = IManager(_manager);
+    }
+
+    /**
+     * @notice Returns the storage, major, minor, and patch version of the contract.
+     * @return Storage version of the contract.
+     * @return Major version of the contract.
+     * @return Minor version of the contract.
+     * @return Patch version of the contract.
+     */
+    function getVersionNumber()
+        external
+        pure
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return (1, 1, 0, 0);
+    }
+
+    /**
+     * @notice Checks if a group meets the validator group health requirements.
+     * @param group The group to check for.
+     * @return Whether or not the group is valid.
+     */
+    function isValidGroup(address group) public view returns (bool) {
+        IValidators validators = getValidators();
+
+        // add check if group is !registered
+        if (!validators.isValidatorGroup(group)) {
+            return false;
+        }
+
+        (address[] memory members, , , , , uint256 slashMultiplier, ) = validators
+            .getValidatorGroup(group);
+
+        // check if group has no members
+        if (members.length == 0) {
+            return false;
+        }
+        // check for recent slash
+        if (slashMultiplier < 10**24) {
+            return false;
+        }
+        // check that at least one member is elected.
+        for (uint256 i = 0; i < members.length; i++) {
+            if (isGroupMemberElected(members[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * @notice Returns expected Celo amount voted for by Account contract
+     * vs actual amount voted for by Acccount contract
+     * @param group The group.
+     */
+    function getExpectedAndRealCeloForGroup(address group) public view returns (uint256, uint256) {
+        bool isSpecificGroupStrategy = specificGroupStrategy.isSpecificGroupStrategy(group);
+        bool isActiveGroup = defaultStrategy.groupsContain(group);
+        uint256 realCelo = account.getCeloForGroup(group);
+
+        if (!isSpecificGroupStrategy && !isActiveGroup) {
+            return (0, realCelo);
+        }
+
+        if (isSpecificGroupStrategy && !isActiveGroup) {
+            return (
+                manager.toCelo(specificGroupStrategy.getTotalStCeloVotesForStrategy(group)),
+                realCelo
+            );
+        }
+
+        if (!isSpecificGroupStrategy && isActiveGroup) {
+            uint256 stCeloSupply = stakedCelo.totalSupply();
+            uint256 stCeloInDefaultStrategy = stCeloSupply -
+                specificGroupStrategy.getTotalStCeloInSpecificGroupStrategies();
+            uint256 supposedStCeloInActiveGroup = stCeloInDefaultStrategy /
+                defaultStrategy.getGroupsLength();
+
+            return (manager.toCelo(supposedStCeloInActiveGroup), realCelo);
```

Comment:
I think I may have mentioned this before, keeping accounting of how much each group gets from DefaultStrategy specifically could clean some things up, this being an example.

Reply:
Accounting for Default group was added